### PR TITLE
ios xr implementation of snmpserver

### PR DESCRIPTION
**Pull request by tphoney**
snmp_server for IOS_XR, contact and location

### CODE REVIEW

File: tests/test_snmpserver.rb [hunk 1]
```diff
@@ -145,23 +176,34 @@ def test_snmpserver_sys_contact_set_special_chars
...
-    contact = line.to_s.gsub('sys contact:', '').strip unless line.nil?
+    if platform == :nexus
+      line = assert_show_match(command: 'show snmp | no-more',
+                               pattern: /sys contact: [^\n\r]*/)
+      contact = ''
+      contact = line.to_s.gsub('sys contact:', '').strip unless line.nil?
+    else
+      refute_show_match(command: 'show running-config snmp-server',
+                        pattern: /snmp-server contact [^\n\r]+/)
+      contact = ''
```
Comment (chburket) [inline tests/test_snmpserver.rb:187]: Is there supposed to be a `contact = line.to_s.gsub`... here?

Comment (tphoney) [inline tests/test_snmpserver.rb:187]: @chburket the iOS_xr does not display contact if it is not set. Therefore there is nothing to gsub.

Comment (chburket) [inline tests/test_snmpserver.rb:187]: Just a nit, but doesn't that mean you are assuming a default value of ''?  If the default value in the yaml ever changes then this test will fail.  Not going to happen, so I'm fine with it as is.

**Comment by chburket** [comment]
:+1:

**Pull request merged**